pages/_app.js

```diff
@@ -3,12 +3,27 @@ import initialProjects from "@/lib/projects.js";
 import useLocalStorageState from "use-local-storage-state";
 import Layout from "@/components/Layout";
 import { nanoid } from "nanoid";
+import { useEffect, useState } from "react";
 
 export default function App({ Component, pageProps }) {
   const [projects, setProjects] = useLocalStorageState("projects", {
     defaultValue: initialProjects,
   });
 
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  function handleToggleForm() {
+    setIsFormOpen(!isFormOpen);
+  }
+
+  useEffect(() => {
+    if (isFormOpen) {
+      document.body.style.overflow = "hidden";
+    } else {
+      document.body.style.overflow = "auto";
+    }
+  }, [isFormOpen]);
+
```

Comment:
If you plan to add more effects in the future, you might want to consider adding a cleanup function to prevent potential issues, 
Additionally, you could refactor the code for better readability `useEffect(() => {
  document.body.style.overflow = isFormOpen ? "hidden" : "auto";
}, [isFormOpen]);`